build issues with LLVM 13.0.0git

LLVM version : 13.0.0 (https://github.com/llvm/llvm-project.git f0ccdde3c9ab23f997a4a30a0e9ab7d9435ec1c6)

Build options : 

  cmake -H. -G Ninja -Bbuild \
  -DBUILD_SHARED_LIBS=ON \
  -DCMAKE_BUILD_TYPE=Release \
  -DCMAKE_INSTALL_PREFIX=/usr \
  -DCMAKE_POSITION_INDEPENDENT_CODE=ON \
  -DCMAKE_SKIP_RPATH=ON \
  -DLLVM_INCLUDE_TESTS=ON \
  -DLLVM_EXTERNAL_LIT=/usr/bin/lit

  ninja -C build
  ninja -C build llvm-spirv

Error :

==> Making package: spirv-llvm-translator-git 13.0.0.2021_03_08.1217.d46ee756-1 (Mon 08 Mar 2021 08:43:22 PM EST)
==> Checking runtime dependencies...
==> Checking buildtime dependencies...
==> Retrieving sources...
  -> Updating SPIRV-LLVM-Translator git repo...
Fetching origin
==> Validating source files with md5sums...
    SPIRV-LLVM-Translator ... Skipped
==> Extracting sources...
  -> Creating working copy of SPIRV-LLVM-Translator git repo...
Cloning into 'SPIRV-LLVM-Translator'...
done.
==> Starting pkgver()...
==> Starting build()...
-- The CXX compiler identification is Clang 13.0.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found ZLIB: /usr/lib/libz.so (found version "1.2.11")
-- Found LibXml2: /usr/lib/libxml2.so (found version "2.9.10")
-- Linker detection: unknown
-- Found LLVM: 13.0.0git
-- Found PkgConfig: /usr/bin/pkg-config (found version "1.7.3")
-- Checking for one of the modules 'SPIRV-Tools'
-- Configuring done
-- Generating done
-- Build files have been written to: /home/kevin/LLVM/git/spirv-llvm-translator-git/src/SPIRV-LLVM-Translator/build
ninja: Entering directory `build'
[14/38] Building CXX object lib/SPIRV/CMakeFiles/LLVMSPIRVLib.dir/LLVMToSPIRVDbgTran.cpp.o
FAILED: lib/SPIRV/CMakeFiles/LLVMSPIRVLib.dir/LLVMToSPIRVDbgTran.cpp.o
/usr/bin/clang++  -m64  -I../include -I../lib/SPIRV -I../lib/SPIRV/libSPIRV -I../lib/SPIRV/Mangler -D_FORTIFY_SOURCE=2 -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -O3 -DNDEBUG -fPIC  -fno-exceptions -std=gnu++14 -MD -MT lib/SPIRV/CMakeFiles/LLVMSPIRVLib.dir/LLVMToSPIRVDbgTran.cpp.o -MF lib/SPIRV/CMakeFiles/LLVMSPIRVLib.dir/LLVMToSPIRVDbgTran.cpp.o.d -o lib/SPIRV/CMakeFiles/LLVMSPIRVLib.dir/LLVMToSPIRVDbgTran.cpp.o -c ../lib/SPIRV/LLVMToSPIRVDbgTran.cpp
../lib/SPIRV/LLVMToSPIRVDbgTran.cpp:104:34: error: no member named 'getVariableLocation' in 'llvm::DbgVariableIntrinsic'
  llvm::Value *Alloca = DbgDecl->getVariableLocation();
                        ~~~~~~~  ^
../lib/SPIRV/LLVMToSPIRVDbgTran.cpp:119:18: error: no member named 'getVariableLocation' in 'llvm::DbgVariableIntrinsic'; did you mean 'getVariableLocationOp'?
  if (!DbgValue->getVariableLocation(/* AllowNullOp = */ false))
                 ^~~~~~~~~~~~~~~~~~~
                 getVariableLocationOp
/usr/include/llvm/IR/IntrinsicInst.h:203:10: note: 'getVariableLocationOp' declared here
  Value *getVariableLocationOp(unsigned OpIdx) const;
         ^
../lib/SPIRV/LLVMToSPIRVDbgTran.cpp:139:26: error: no member named 'getVariableLocation' in 'llvm::DbgVariableIntrinsic'; did you mean 'getVariableLocationOp'?
  Value *Val = DbgValue->getVariableLocation(/* AllowNullOp = */ false);
                         ^~~~~~~~~~~~~~~~~~~
                         getVariableLocationOp
/usr/include/llvm/IR/IntrinsicInst.h:203:10: note: 'getVariableLocationOp' declared here
  Value *getVariableLocationOp(unsigned OpIdx) const;
         ^
3 errors generated.
[36/38] Building CXX object lib/SPIRV/CMakeFiles/LLVMSPIRVLib.dir/SPIRVWriter.cpp.o
ninja: build stopped: subcommand failed.
==> ERROR: A failure occurred in build().
    Aborting...

--- a/lib/SPIRV/LLVMToSPIRVDbgTran.cpp
+++ b/lib/SPIRV/LLVMToSPIRVDbgTran.cpp
@@ -101,7 +101,7 @@ void LLVMToSPIRVDbgTran::finalizeDebugDeclare(
     return;
   SPIRVExtInst *DD = static_cast<SPIRVExtInst *>(V);
   SPIRVBasicBlock *BB = DD->getBasicBlock();
-  llvm::Value *Alloca = DbgDecl->getVariableLocation();
+  llvm::Value *Alloca = DbgDecl->getVariableLocationOp(0);
 
   using namespace SPIRVDebug::Operand::DebugDeclare;
   SPIRVWordVec Ops(OperandCount);
@@ -116,7 +116,7 @@ void LLVMToSPIRVDbgTran::finalizeDebugDeclare(
 
 SPIRVValue *LLVMToSPIRVDbgTran::createDebugValuePlaceholder(
     const DbgVariableIntrinsic *DbgValue, SPIRVBasicBlock *BB) {
-  if (!DbgValue->getVariableLocation(/* AllowNullOp = */ false))
+  if (!DbgValue->getVariableLocationOp(0))
     return nullptr; // It is pointless without new value
 
   DbgValueIntrinsics.push_back(DbgValue);
@@ -136,7 +136,7 @@ void LLVMToSPIRVDbgTran::finalizeDebugValue(
     return;
   SPIRVExtInst *DV = static_cast<SPIRVExtInst *>(V);
   SPIRVBasicBlock *BB = DV->getBasicBlock();
-  Value *Val = DbgValue->getVariableLocation(/* AllowNullOp = */ false);
+  Value *Val = DbgValue->getVariableLocationOp(0);
 
   using namespace SPIRVDebug::Operand::DebugValue;
   SPIRVWordVec Ops(MinOperandCount);

